src/service.rs

```diff
@@ -73,11 +73,12 @@ impl ServiceRequest {
         mut req: HttpRequest,
         pl: Payload,
     ) -> Result<Self, (HttpRequest, Payload)> {
-        if Rc::strong_count(&req.0) == 1 && Rc::weak_count(&req.0) == 0 {
```

Comment:
keep the weak count as a debug assertion here and on from_request

Reply:
`Rc::get_mut` only work on this condition. There is no need to double check.

Reply:
`Rc::get_mut` only work on this condition. `from_request` indeed worth a debug assert